atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraMutationTimestampProvider.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cassandra;
+
+import com.google.common.collect.Multimap;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+
+/**
+ * This interface produces Cassandra timestamps for mutations, where it makes sense for these to be overridden.
+ * Generally, to ensure that Cassandra is able to compact tombstones smoothly, these values should increase over
+ * time and should largely be in line with AtlasDB's notion of logical time.
+ *
+ * Calling these methods must be cheap, at least on an amortised basis; they may be called multiple times
```

Comment:
With this PR, can you let me know how often the three methods below will be called, including any higher level batching or extra uses for fetched timestamps?

Reply:
So beneath this level, I've added it to the docs.

Higher level batching takes place in `RequestBatchingTimestampService`, where a given Atlas client will always maintain only one open timestamp request to timelock.

Reply:
@gmaretic we may need to do follow up work here or at least make sure we understand how much overhead the timestamp call is adding to each ranged tombstone. If this slows targeted sweep down by a factor of 2-3x, we should definitely revisit.